jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -152,13 +193,465 @@ default int capacity()
     }
 
     /**
+     * @return the number of bytes left for appending in the {@code ByteBuffer}
+     */
+    default int space()
+    {
+        return capacity() - remaining();
+    }
+
+    /**
+     * @return whether the {@code ByteBuffer} has remaining bytes left for appending
+     */
+    default boolean isFull()
+    {
+        return space() == 0;
+    }
+
+    /**
+     * Clears the contained byte buffer to be empty in flush mode.
      * @see BufferUtil#clear(ByteBuffer)
      */
     default void clear()
     {
         BufferUtil.clear(getByteBuffer());
     }
 
+    /**
+     * Copies the contents of the given byte buffer at the end of this buffer.
+     * @param bytes the byte buffer to copy from.
+     * @return true if all bytes of the given buffer were copied, false otherwise.
+     * @see BufferUtil#append(ByteBuffer, ByteBuffer)
+     */
+    default boolean append(ByteBuffer bytes)
```

Comment:
I don't think this is a great idea.
This API is never used by users of RBB.
Its only existence is for users of `Aggregator`.
Not even `Accumulator` uses this API.

As such, I think it should just belong to `Aggregator`, and be removed from the other 2 classes, which have different semantic, and trying to conflate all semantics into one class is not a good idea.